src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -0,0 +1,234 @@
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            status = True
+        elif "host_storage_map" in result.keys():
+            for value in list(result['host_storage_map'].values()):
+                if value['storage']['smd_info']['devices']:
+                    for device in value['storage']['smd_info']['devices']:
+                        if device['dev_state'] == dev_state and device['led_state'] == led_state:
+                            status = True
+        else:
+            self.log.info("##Unsupported result dictionary %s", result)
+        return status
+
+    def run_ior_with_nvme_fault(self, oclass=None):
+        """Perform ior write with nvme fault testing.
+
+        Test steps:
+            1. Start the DAOS servers.
+            2. Verify the drive status LED.
+            3. Create the pool and container with RF.
+            4. While IOR (EC+GX) in progress put single drive to fault mode.
+            5. IOR will halt until rebuild is finished.
+            6. IOR should complete without any error.
+            7. Drive status LED should be solid Amber ON indicating it’s faulty.
+            8. Replace the same drive back.
+            9. Drive status LED should be off indicating good device is plugged-in.
+           10. Reset and cleanup.
+        Args:
+            oclass (str, optional): object class (eg: RP_2G8, S1,etc). Defaults to None
+        """
+        # 1. Bring up server and reset led of all devices
+        self.log_step("Bring up server and reset led of all devices.")
+        if oclass is None:
+            oclass = self.ior_cmd.dfs_oclass.value
+        device_info = get_storage_query_device_info(self, self.dmg)
+        devices = [device['uuid'] for device in device_info]
+        self.log.info("==device_info= %s", devices)
+        test_dev = devices[0]
+        for device in devices:
+            self.run_vmd_led_identify(device, reset=True)
+
+        # 2. Check drive status led state
+        self.log_step("Check drive status led state.")
+        err_dev = []
+        for device in devices:
+            if not self.verify_dev_led_state(device):
+                err_dev.append(device)
+        if err_dev:
+            self.fail("#Device {} not in expected NORMAL, OFF state".format(err_dev))
+
+        # 3. Create the pool and container with RF and start IOR
+        self.log_step("Create the pool and container with RF and start IOR")
+        self.add_pool(connect=False)
+        self.add_container(self.pool)
+
+        job_manager = get_job_manager(self, subprocess=None, timeout=120)
+        thread_queue = Queue()
+
+        ior_test_seq = self.params.get("ior_test_sequence", "/run/ior/iorflags/*")[0]
+        thread = []
+        kwargs = {"thread_queue": thread_queue, "job_id": 1}
+        ior_kwargs = {
+            "test": self,
```

Comment:
Why not define a single kwargs
```suggestion
        kwargs = {
            "thread_queue": thread_queue,
            "job_id": 1,
            "test": self,
```

Reply:
Ah, I see now. Thanks